hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/service/HiveToGlueTransformer.java

```diff
@@ -0,0 +1,170 @@
+        .withRetention(table.getRetention())
+        .withStorageDescriptor(sd)
+        .withTableType(table.getTableType());
+  }
+
+  public PartitionInput transformPartition(final Partition partition) {
+    final Date date = convertTableDate(partition.getLastAccessTime());
+
+    final org.apache.hadoop.hive.metastore.api.StorageDescriptor storageDescriptor = partition.getSd();
+    final Collection<Column> columns = extractColumns(storageDescriptor.getCols());
+
+    final SerDeInfo glueSerde = new SerDeInfo()
+        .withName(storageDescriptor.getSerdeInfo().getName())
+        .withParameters(storageDescriptor.getSerdeInfo().getParameters())
+        .withSerializationLibrary(storageDescriptor.getSerdeInfo().getSerializationLib());
+
+    final List<Order> sortOrders = extractSortOrders(storageDescriptor.getSortCols());
+
+    final StorageDescriptor sd = new StorageDescriptor()
+        .withBucketColumns(storageDescriptor.getBucketCols())
+        .withColumns(columns)
+        .withCompressed(storageDescriptor.isCompressed())
+        .withInputFormat(storageDescriptor.getInputFormat())
+        .withLocation(storageDescriptor.getLocation())
+        .withNumberOfBuckets(storageDescriptor.getNumBuckets())
+        .withOutputFormat(storageDescriptor.getOutputFormat())
+        .withParameters(storageDescriptor.getParameters())
+        .withSerdeInfo(glueSerde)
+        .withSortColumns(sortOrders)
+        .withStoredAsSubDirectories(storageDescriptor.isStoredAsSubDirectories());
+
+    return new PartitionInput()
+        .withLastAccessTime(date)
+        .withParameters(partition.getParameters())
+        .withStorageDescriptor(sd)
+        .withValues(partition.getValues());
+  }
+
+  public String glueDbName(String dbName) {
+    return (gluePrefix == null) ? dbName : gluePrefix + dbName;
+  }
+
+  public String glueDbName(Table table) {
+    return glueDbName(table.getDbName());
+  }
+
+  private List<Order> extractSortOrders(final List<org.apache.hadoop.hive.metastore.api.Order> hiveOrders) {
+    final List<Order> sortOrders = new ArrayList<>();
+    if (hiveOrders == null) {
+      return sortOrders;
+    }
+
+    for (final org.apache.hadoop.hive.metastore.api.Order hiveOrder : hiveOrders) {
+      final Order order = new Order().withSortOrder(hiveOrder.getOrder()).withColumn(hiveOrder.getCol());
+      sortOrders.add(order);
+    }
+    return sortOrders;
+  }
+
+  private List<Column> extractColumns(final List<FieldSchema> colList) {
+    final List<Column> columns = new ArrayList<>();
+    if (colList == null) {
+      return columns;
+    }
+
+    for (final FieldSchema fieldSchema : colList) {
+      final Column col = new Column()
+          .withName(fieldSchema.getName())
+          .withType(fieldSchema.getType())
+          .withComment(fieldSchema.getComment());
+
+      columns.add(col);
+    }
+    return columns;
+  }
+
+  private Date convertTableDate(final Integer lastAccessTime) {
```

Comment:
Should the method name be `convertMillisToTableDate`? You're missing the target in the name.